agent/src/util/device_plugin_service.rs

```diff
@@ -300,6 +302,16 @@ impl DevicePluginService {
                     device_usage_id.clone(),
                 );
 
+                let hash_id_value = generate_digest(&device_usage_id, 3);
```

Comment:
I wonder if we should `to_upper` this so the env vars are still all caps. This is purely aesthetic probably

Reply:
call to_uppercase() so environment variables remain all caps.